bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -650,8 +650,18 @@ impl Connection {
     }
 
     /// Attempts to flush any data previously buffered by a call to [send](`Self::poll_send`).
+    ///
+    /// poll_flush can only flush data that s2n-tls has already encrypted and
+    /// buffered for sending. poll_send may need to be called again to fully send
+    /// all data. See the [Usage Guide](https://github.com/aws/s2n-tls/blob/main/docs/usage-guide/topics/ch07-io.md)
+    /// for more details.
     pub fn poll_flush(&mut self) -> Poll<Result<&mut Self, Error>> {
```

Comment:
Should poll_flush update send_pending?
https://github.com/aws/s2n-tls/blob/9cd117f3d9a94912d5e8f2a61ffbfd31f11d2946/bindings/rust/s2n-tls/src/renegotiate.rs#L183

Reply:
Nope. As the comment mentions, poll_flush successfully completing doesn't mean that the previous send successfully completed. See the test I wrote.

The send buffer isn't the only send state kept. It's just the easiest / most straight forward state to check when you want to check whether there's a partial send in progress. So even if we flush the buffer, we still don't want to renegotiate because we'll wipe the rest of the send state.

Although as I type that out... that makes surfacing poll_flush in the bindings a little dangerous. You could "trick" renegotiation by clearing the send buffer. However, I think you'd still be fine specifically because we also track send_pending.